backends/advanced/src/advanced_omi_backend/controllers/system_controller.py

```
@@ -455,3 +455,103 @@ async def set_memory_provider(provider: str):
+        with open(config_path, 'w') as f:
+            yaml.dump(full_config, f, default_flow_style=False, allow_unicode=True)
+
+        # Reload config in memory (hot-reload)
+        load_models_config(force_reload=True)
+
+        logger.info("Chat configuration updated successfully")
+
+        return {"success": True, "message": "Chat configuration updated successfully"}
+
+    except Exception as e:
+        logger.error(f"Error saving chat config: {e}")
```

Ruff (0.14.10)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)